index.js

```diff
@@ -34,350 +46,427 @@ client.on("ready", () => {
-			await validate(false);
+			outcomes = outcomes.trim();
+			response.push(`Title: ${data.title}`);
+			response.push(`Prediction-ID: ${data.id}`);
+			response.push(`Broadcaster: ${data.broadcaster_name}`);
+			response.push(`Outcomes:\n${outcomes}`);
+			response.push(`Prediction Window: ${data.prediction_window} seconds`);
+			response.push(`Prediction Status: ${data.status}`);
+			response.push(`Created At: ${toDiscordTimestamp(data.created_at)}`);
+		}
+		await interaction.editReply({
+			content: response.join("\n")
 		});
-	}).catch((err) => {
-		interaction.editReply({content: 'Token validation/refresh failed!'});
+	}).catch(async (err) => {
+		await interaction.editReply({
+			content: `Error creating prediction on Twitch: ${err}`
+		});
+		await validate(false);
 	});
 }
 
-function endPrediction(interaction) {
+// https://dev.twitch.tv/docs/api/reference#end-prediction
+async function endPrediction(interaction) {
+	const broadcasterId = await getBroadcasterId();
 	let status = interaction.options.getString('status');
 	status = status.substring(0, status.indexOf(' ')).trim();
-	let winning_outcome_id = interaction.options.getString('winning_outcome_id');
-	validate(false).then(async (value) => {
-		fetch('https://api.twitch.tv/helix/predictions', {
-			method: 'PATCH',
-			headers: {
-				'Client-ID': process.env.TWITCH_CLIENT_ID,
-				'Authorization': `Bearer ${tokens.access_token}`,
-				'Content-Type': 'application/json'
-			},
-			body: JSON.stringify({
-				broadcaster_id: (await fetch(`https://api.twitch.tv/helix/users?login=${process.env.BROADCASTER_LOGIN}`, {
-					headers: {
-						'Client-ID': process.env.TWITCH_CLIENT_ID,
-						'Authorization': `Bearer ${tokens.access_token}`
-					}
-				}).then(res => res.json()).catch(err => console.error)).data[0].id,
-				id: interaction.options.getString('id'),
-				status: status,
-				winning_outcome_id: (winning_outcome_id ? winning_outcome_id : undefined)
-			})
-		}).then(res => res.json()).then(res => {
-			let response = 'Prediction successfully ended!\n';
-			if (res.error) {
-				response = `Error: ${res.error}\nError-Message: ${res.message}`;
-			} else {
-				let data = res.data[0];
-				let outcomes = '';
-				for (let i = 0; i < data.outcomes.length; i++) {
-					let outcome = data.outcomes[i];
-					outcomes += `> ${outcome.title}\n> > Outcome-ID: ${outcome.id}\n> > Users: ${outcome.users}\n> > Channel Points: ${outcome.channel_points}\n> > Color: ${outcome.color}\n`;
-					outcomes += '> > Top Predictors:\n';
-					for (let j = 0; outcome.top_predictors && j < outcome.top_predictors.length; j++) {
-						let topPredictor = outcome.top_predictors[j].user;
-						outcomes += `> > > User: ${topPredictor.name} (${topPredictor.id})\n> > > > Channel Points used: ${topPredictor.channel_points_used}\n> > > > Channel Points won: ${topPredictor.channel_points_won}\n`;
-					}
+	const winningOutcomeId = interaction.options.getString('winning_outcome_id');
```

Comment:
Use null coalising operator to not having to do an nullable check a few lines below this one.